src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisWebhookService.java

```diff
@@ -129,40 +136,52 @@ public String deleteLectureFromPyrisDB(List<AttachmentUnit> attachmentUnits) {
     /**
      * adds the lectures to the vector database on pyris
      *
-     * @param attachmentUnits The attachmentUnit that got Updated / erased
+     * @param attachmentUnit The attachmentUnit that got Updated / erased
      * @return jobToken if the job was created
      */
-    public String addLectureUnitsToPyrisDB(List<AttachmentUnit> attachmentUnits) {
-        if (!lectureIngestionEnabled(attachmentUnits.getFirst().getLecture().getCourse())) {
-            return null;
-        }
-        try {
-            List<PyrisLectureUnitWebhookDTO> toUpdateAttachmentUnits = new ArrayList<>();
-            attachmentUnits.stream().filter(unit -> unit.getAttachment().getAttachmentType() == AttachmentType.FILE && unit.getAttachment().getLink().endsWith(".pdf"))
-                    .forEach(unit -> {
-                        toUpdateAttachmentUnits.add(processAttachmentForUpdate(unit));
-                    });
-            if (!toUpdateAttachmentUnits.isEmpty()) {
-                return executeLectureWebhook(toUpdateAttachmentUnits);
+    public String addLectureUnitToPyrisDB(AttachmentUnit attachmentUnit) {
+        if (lectureIngestionEnabled(attachmentUnit.getLecture().getCourse())) {
+            try {
+                if (attachmentUnit.getAttachment().getAttachmentType() == AttachmentType.FILE && attachmentUnit.getAttachment().getLink().endsWith(".pdf")) {
+                    return executeLectureAdditionWebhook(processAttachmentForUpdate(attachmentUnit));
+                }
+            }
+            catch (Exception e) {
+                log.error(e.getMessage());
```

Comment:
Please do not log the message alone, since the stack trace or the attachmentUnitId can also help finding the problem later on